src/main/scala/org/hammerlab/guacamole/Serialization.scala

```diff
@@ -19,27 +19,83 @@
+import org.hammerlab.guacamole.reference.ReferenceBroadcast.MapBackedReferenceSequence
 import org.hammerlab.guacamole.variants._
 
-class GuacamoleKryoRegistrator extends ADAMKryoRegistrator {
-  override def registerClasses(kryo: Kryo) {
-    // For debugging, it can be helpful to uncomment the following line. This will raise an error if we try to serialize
-    // anything without a registered Kryo serializer.
-    // kryo.setRegistrationRequired(true)
+import scala.collection.mutable.ArrayBuffer
 
-    // This allows us to serialize object graphs with cycles. It should default to true, so kind of strange that we have
-    // to set it, but without this line we see infinite recursion.
-    kryo.setReferences(true)
+class GuacamoleKryoRegistrator extends KryoRegistrator {
+  override def registerClasses(kryo: Kryo) {
 
     // Register ADAM serializers.
-    super.registerClasses(kryo)
+    new ADAMKryoRegistrator().registerClasses(kryo)
+
+    kryo.register(classOf[Array[BDGGenotype]])
+    kryo.register(classOf[SequenceDictionary])
+    kryo.register(classOf[SequenceRecord])
+
+    kryo.register(classOf[PileupElement])
+    kryo.register(classOf[Array[PileupElement]])
+
+    kryo.register(classOf[NormalDNASingleSampleSingleAlleleEvidence])
+    kryo.register(classOf[TumorDNASingleSampleSingleAlleleEvidence])
+
+    // These are needed for MultiSampleSingleAlleleEvidence; perTumorDnaSampleTopMixtures and probably other fields.
+    kryo.register(Class.forName("scala.collection.immutable.MapLike$$anon$2"))  // MapLike.MappedValues?
+    kryo.register(Class.forName("org.hammerlab.guacamole.commands.jointcaller.evidence.MultiSampleSingleAlleleEvidence$$anonfun$16"))
+    kryo.register(Class.forName("org.hammerlab.guacamole.commands.jointcaller.evidence.MultiSampleSingleAlleleEvidence$$anonfun$10"))
+
+    kryo.register(classOf[Parameters])
+    kryo.register(Class.forName("scala.Enumeration$Val"))
+    kryo.register(SomaticGenotypePolicy.getClass)
+    kryo.register(classOf[MultiSampleAnnotations])
+    kryo.register(classOf[InputCollection])
+    kryo.register(classOf[Input])
+    kryo.register(Input.Analyte.getClass)
+    kryo.register(Input.TissueType.getClass)
+    kryo.register(classOf[InsufficientNormal])
+    kryo.register(classOf[TumorRNASingleSampleSingleAlleleEvidence])
+    kryo.register(Class.forName("org.apache.spark.broadcast.TorrentBroadcast"))
+    kryo.register(classOf[BroadcastBlockId])
+    kryo.register(classOf[Parameters.SomaticGenotypePolicy.Value])
+    kryo.register(classOf[StrandBias])
+    kryo.register(classOf[SingleSampleAnnotations])
+    kryo.register(classOf[AlleleAtLocus])
+    kryo.register(classOf[MultiSampleSingleAlleleEvidence])
+    kryo.register(classOf[MultiSampleMultiAlleleEvidence])
+    kryo.register(classOf[Array[MultiSampleMultiAlleleEvidence]])
+
+    kryo.register(classOf[Pileup])
```

Comment:
another suspicious one -- we shouldn't be serializing these as @arahuja mentioned. ok by me though to address that in a separate PR